opencensus/trace/span.py

```diff
@@ -281,16 +387,26 @@ def format_span_json(span):
         span_json['parentSpanId'] = parent_span_id
 
     if span.attributes:
-        span_json['attributes'] = attributes.Attributes(
+        span_json['attributes'] = attributes_module.Attributes(
             span.attributes).format_attributes_json()
 
     if span.stack_trace is not None:
         span_json['stackTrace'] = span.stack_trace.format_stack_trace_json()
 
-    if span.time_events:
+    formatted_time_events = []
```

Comment:
Not sure why we'd want to combine annotations and message events?

Reply:
Your guess is as good as mine, I did this to preserve the old behavior.

Reply:
Actually `format_span_json` doesn't have any callers, and neither do the other json formatting methods it calls. `format_legacy_trace_json` is only used by the logging and file exporters. I'll remove them in another PR.

Reply:
SGTM

Reply:
The logging and file exporters only incidentally call `format_legacy_trace_json`, the real reason for this odd formatting is stackdriver. It looks like the reason annotations and message events were combined is because that's how they're [reported to stackdriver](https://cloud.google.com/trace/docs/reference/v2/rest/v2/TimeEvents#TimeEvent).

The stackdriver data model leaks out into other classes too. Attributes, links, stack traces, and etc. have formatting methods specific to stackdriver, and we're using these methods to serialize `SpanData`s for the logging and file exporters.

I'm moving the stackdriver-specific formatting into the stackdriver exporter in #651 and adding generic serialization methods for the exporters that need them.